starlette/routing.py

```diff
@@ -273,7 +275,25 @@ async def handle(self, scope: Scope, receive: Receive, send: Send) -> None:
                 )
             await response(scope, receive, send)
         else:
-            await self.app(scope, receive, send)
+            bytes_read = 0
+            max_body_size = self.request_max_size or scope.get("request_max_size")
+
+            async def receive_wrapper() -> Message:
+                nonlocal bytes_read
+                message = await receive()
+                if message["type"] != "http.request" or max_body_size is None:
+                    return message
+
+                body = message.get("body", b"")
+                bytes_read += len(body)
+                if bytes_read > max_body_size:
+                    raise HTTPException(
```

Comment:
Why don't we make this a middleware since we have that capability? It would help keep the routing code more routing focused.

Reply:
The usual setup is - a global limit and override per route that can accept a large payload. Usually, you don't have lots of them.

Middleware is an option too. However, it requires a Mount route type.

Reply:
Ah right I forgot that we only added middleware to Mount routes. Maybe this is a good argument for adding it to every route.

Reply:
Yes, per-route middleware is good for core reuse but not sure if it is a good DX.